cub/test/catch2_test_device_radix_sort_keys.cu

```diff
@@ -460,3 +460,70 @@ CUB_TEST("DeviceRadixSort::SortKeys: DoubleBuffer API", "[keys][radix][sort][dev
 
   REQUIRE(ref_keys == keys);
 }
+
+template <typename key_t, typename num_items_t>
+void do_large_offset_test(std::size_t num_items)
+{
+  const bool is_descending = GENERATE(false, true);
+
+  CAPTURE(num_items, is_descending);
+
+  try
+  {
+    large_array_sort_helper<key_t> arrays;
+    arrays.initialize_for_unstable_key_sort(num_items, is_descending);
+
+    TIME(c2h::cpu_timer timer);
+
+    double_buffer_sort_t action(is_descending);
+    action.initialize();
+    const num_items_t typed_num_items = static_cast<num_items_t>(num_items);
+    launch(action, arrays.keys_buffer, typed_num_items, begin_bit<key_t>(), end_bit<key_t>());
+
+    arrays.keys_buffer.selector = action.selector();
+    action.finalize();
+
+    auto& sorted_keys = arrays.keys_buffer.selector == 0 ? arrays.keys_in : arrays.keys_out;
+
+    TIME(timer.print_elapsed_seconds_and_reset("Device sort"));
+
+    arrays.verify_unstable_key_sort(num_items, is_descending, sorted_keys);
+  }
+  catch (std::bad_alloc& e)
+  {
+    const std::size_t num_bytes = num_items * sizeof(key_t);
+    std::cerr
+      << "Skipping radix sort test with " << num_items << " elements (" << num_bytes << " bytes): " << e.what() << "\n";
```

Comment:
suggestion: you could use `WARN` instead.
question: There should be no output from tests other than in case of an error. Logging is not visible in successful runs of CI. Extra error messages in failed runs increase mental load. Other than for tests development, what purpose this logging serves?

Reply:
I've been wondering this myself. It seems like something that we should have *some* notification for, but I agree that this current approach isn't actionable or useful.

Perhaps later we can make sure that all tests fit within X GiB of device memory and enable warnings/failures here when running on hardware that we know should be able to handle the allocations, just to make sure that these tests are exercised at some point in our test cycle.

But for now, I'll guard these outputs behind a debug macro so they won't be so noisy.